### PR TITLE
chore: create Nameplate component

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

WEB-779

Replace `<GroupAvatar>` with `<Nameplate>`.

Change `<UserBadge>` to be a wrapper that adds user-specific logic to `<Nameplate>`.

Replace `<UserAvatarCard>` with `<UserBadge>`.

## Test Plan

No specific feature testing, just regression for displaying user/group names (Navigation bar, User management, Workspace members)

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/kit/UserAvatar.tsx [hunk 1]
```diff
@@ -2,11 +2,9 @@ import React from 'react';
 
 import Avatar, { Props as AvatarProps } from 'shared/components/Avatar';
 import useUI from 'shared/contexts/stores/UI';
-import { DetailedUser } from 'types';
-import { getDisplayName } from 'utils/user';
-
+import { getDisplayName, UserNameFields } from 'utils/user';
 export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
-  user?: DetailedUser;
+  user?: UserNameFields;
```
Comment (EmilyBonar) [inline webui/react/src/components/kit/UserAvatar.tsx:7]: question: Why was this changed? It seems like it causes ripple effects requiring typecasts elsewhere.

File: webui/react/src/components/kit/UserBadge.tsx [hunk 2]
```diff
@@ -1,32 +1,25 @@
...
-import { getDisplayName, UserNameFields } from 'utils/user';
+import { UserNameFields } from 'utils/user';
 
-import css from './UserBadge.module.scss';
+import Nameplate from './Nameplate';
+import UserAvatar from './UserAvatar';
 
-export interface Props extends Omit<AvatarProps, 'darkLight' | 'displayName'> {
+export interface Props {
+  className?: string;
```
Comment (EmilyBonar) [inline webui/react/src/components/kit/UserBadge.tsx:9]: fix: again, shouldn't expose `className`

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Mostly looks good, but we should stop exposing `className` in UI Kit components

File: webui/react/src/components/kit/UserBadge.tsx [hunk 3]
```diff
@@ -1,32 +1,25 @@
...
 
-const UserBadge: React.FC<Props> = ({ user, compact, ...rest }) => {
-  const { ui } = useUI();
-  const displayName = getDisplayName(user);
-
-  const avatar = <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
-  const classnames = [css.avatarCard];
-  if (compact) classnames.push(css.compact);
-
+const UserBadge: React.FC<Props> = ({ user, compact, className }) => {
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/UserBadge.tsx:14]: We should remove UserBadge from the UI kit since it's just a thin wrapper around Nameplate.

File: webui/react/src/pages/DesignKit.tsx [hunk 4]
```diff
@@ -882,6 +885,53 @@ const UserBadgeSection: React.FC = () => {
...
+      </AntDCard>
+      <AntDCard title="Usage">
+        <li>With name and alias</li>
+        <Nameplate
+          alias={testUser.displayName}
+          icon={<UserAvatar user={testUser} />}
+          name={testUser.username}
+        />
+        <li>Compact format</li>
+        <Nameplate
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:920]: You're missing a case here where it's compact with no alias.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]
I love this. Great job with the name as well, really does a good job indicating what it is. I have a bit of feedback in addition to what Emily mentioned.

File: webui/react/src/components/kit/UserAvatar.tsx (hunk 1 above)
Comment (johnkim-det) [reply webui/react/src/components/kit/UserAvatar.tsx:7]: Thanks for pointing this out. The typecasting I added was actually unnecessary, so I cleaned that up. Apart from that mistake, the `User` type has the most precise subset of fields we need here, so switching to that should minimize the amount of typecasting needed.

**Review by EmilyBonar — APPROVED** [review]
Thanks for making those changes, looks good!

File: webui/react/src/components/kit/UserBadge.tsx [hunk 5]
```diff
@@ -1,32 +1,24 @@
...
 
-const UserBadge: React.FC<Props> = ({ user, compact, ...rest }) => {
-  const { ui } = useUI();
-  const displayName = getDisplayName(user);
-
-  const avatar = <Avatar {...rest} darkLight={ui.darkLight} displayName={displayName} />;
-  const classnames = [css.avatarCard];
-  if (compact) classnames.push(css.compact);
-
+const UserBadge: React.FC<Props> = ({ user, compact, hideAvatarTooltip }) => {
```
Comment (ClaireNeveu) [inline webui/react/src/components/kit/UserBadge.tsx:14]: Does this really need to be a component anymore? If so let's move it out of `kit`.

File: webui/react/src/pages/DesignKit.tsx [hunk 6]
```diff
@@ -990,26 +991,36 @@ const UserAvatarSection: React.FC = () => {
...
   return (
-    <ComponentSection id="UserBadge" title="UserBadge">
+    <ComponentSection id="Nameplate" title="Nameplate">
       <AntDCard>
         <p>
-          A (<code>{'<UserBadge>'}</code>) fully represents a user with a UserAvatar circle icon,
-          and the user&apos;s display name and username. If there is a display name, it appears
-          first, otherwise only the username is visible. A &apos;compact&apos; option reduces the
-          size of the name for use in a smaller form or modal.
+          A (<code>{'<Nameplate>'}</code>) displays identifying information for a user or group. It
```
Comment (ClaireNeveu) [inline webui/react/src/pages/DesignKit.tsx:1001]: Doesn't necessarily display information about a user or group.

**Review by ClaireNeveu — CHANGES_REQUESTED** [review]

File: webui/react/src/components/kit/UserBadge.tsx (hunk 5 above)
Comment (johnkim-det) [reply webui/react/src/components/kit/UserBadge.tsx]: Moving it out of `/kit` for now. 

Actually originally had this with `<UserBadge>` removed, but I ended up making it a wrapper around `<Nameplate>` since it does reduce duplication and it's consistent with `<UserAvatar>`, which is also just a user-specific wrapper around `<Avatar>`. Definitely open to removing it and going with passing the icon and text props directly, though.

**Review by ClaireNeveu — APPROVED** [review]
